File: lib/core_services/comic/interfaces/comic_carousel_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:hoyomi/core_services/interfaces/genre.dart';
import 'package:hoyomi/core_services/interfaces/o_image.dart';

part 'comic_carousel_item.freezed.dart';
part 'comic_carousel_item.g.dart';

@freezed
sealed class ComicCarouselItem with _$ComicCarouselItem {
  const factory ComicCarouselItem({
    String? subText,
    required OImage image,
    required String comicId,
    required String name,
    String? originalName,
    String? type,
    String? episodeDuration,
    DateTime? updatedAt,
    String? quality,
    int? countSub,
    int? countDub,
    double? rate,
    String? notice,
    String? year,
    String? description,
    String? studio,
    String? duration,
    String? language,
    List<Genre>? genres,
    List<Genre>? actors,
  }) = _ComicCarouselItem;

  factory ComicCarouselItem.fromJson(Map<String, dynamic> json) =>
      _$ComicCarouselItemFromJson(json);
}

File: lib/core_services/comic/interfaces/comic_carousel_item.ts
// ...existing imports...
import { OImage } from "../../interfaces/o_image"
import { Genre } from "../../interfaces/genre"

export interface ComicCarouselItem {
  subText?: string
  image: OImage
  comicId: string
  name: string
  originalName?: string
  type?: string
  episodeDuration?: string
  updatedAt?: Date
  quality?: string
  countSub?: number
  countDub?: number
  rate?: number
  notice?: string
  year?: string
  description?: string
  studio?: string
  duration?: string
  language?: string
  genres?: Genre[]
  actors?: Genre[]
}
